scripts/test/system.sh

```diff
@@ -191,3 +192,14 @@ _subctl cloud prepare generic --kubecontext cluster1 && exit 1
 
 _subctl uninstall -y --context cluster2
 _subctl uninstall -y --kubeconfig "${KUBECONFIGS_DIR}"/kind-config-cluster1
+
+sleep 30s # Give time for uninstall to finish
```

Comment:
I think there’s value in having a well-identified `subctl uninstall` test; and if the `recover-brokerinfo` tests rely on `uninstall`, then the latter should be tested first, so that a breakage in `uninstall` doesn’t result in wasted time testing `recover-brokerinfo`.

Avoiding `sleep 30` would be great, but I’m not sure there’s a good way of doing that. `kubectl` can be asked to wait for certain states, perhaps that would work, if there’s an object which gives a reliable indication that the uninstallation is complete.

Reply:
> It's best to avoid hard-coded sleeps. We should wait for the broker and/or submariner namespaces are deleted.
> 
> That said, perhaps we can move some of the tests above the uninstall tests on L193 and simplify to avoid uninstalling and re-installing, which may significantly increase the run time. So prior to L193, we can test the cases where submariner is on same cluster as the broker. eg:
> 
> Expect success:
> 
> ```
> _subctl recover-broker-info --namespace submariner-k8s-broker --context cluster1
> _subctl recover-broker-info --brokercontext cluster1
> _subctl recover-broker-info --brokernamespace submariner-k8s-broker --brokercontext cluster1
> ```
> 
> To verify, we can save off the original `broker-info.subm` and compare with the recovered one.
> 
Thats what I tried to do with `[no_broker_found_error()](https://github.com/submariner-io/subctl/pull/467/files#diff-637916d64af30a6e6f9d31f415d8250201b56786188c35a41c2ee0763bcffb84R43)`. Dind't work as expected so I have changed the implementation.

> Expect failures:
> 
> ```
> _subctl recover-broker-info --namespace non-existent --context cluster1
> _subctl recover-broker-info --namespace submariner-k8s-broker --context cluster2
> _subctl recover-broker-info --brokernamespace non-existent --brokercontext cluster1
> _subctl recover-broker-info --brokernamespace submariner-k8s-broker --brokercontext cluster2
> ```
> 
> For these, verify a non-zero exit code.
> 
> After both clusters are uninstalled on L194, we can re-deploy the broker on cluster1 and test the case where there's no joined clusters and it generates a new IPsec secret:
> 
> ```
> _subctl recover-brokerinfo  --brokercontext cluster1
> _subctl recover-broker-info --namespace submariner-k8s-broker --context cluster1
> ```
> 
> For the broker-only install case, we can join cluster2:
> 
> `_subctl recover-broker-info --brokercontext cluster1 --context cluster2`